File: solutions/java/knapsack/1/src/main/java/Knapsack.java
import java.util.List;

class Knapsack {

    public int maximumValue(int maxCapacity, List<Item> items) {
        int[] maxValuesAtCapacities = new int[maxCapacity + 1];

        for (Item item : items) {
            for (int currentCapacity = maxCapacity; currentCapacity >= item.weight; currentCapacity--) {
                int remainingCapacity = currentCapacity - item.weight;
                int valueWithNewItem = maxValuesAtCapacities[remainingCapacity] + item.value;
                maxValuesAtCapacities[currentCapacity] = Math.max(maxValuesAtCapacities[currentCapacity], valueWithNewItem);
            }
        }

        return maxValuesAtCapacities[maxCapacity];
    }

}

File: solutions/python/knapsack/1/knapsack.py
def maximum_value(maximum_weight, items):
    combinations = [(0, 0)]
    for item in items:
        combinations += [
            (weight + item["weight"], value + item["value"])
            for weight, value in combinations
        ]

    return max(
        value for weight, value in combinations if weight <= maximum_weight
    )
